File: src/main/kotlin/inflearn/java/_1_11_문자열_압축.java
package inflearn.java;

import java.util.Scanner;

public class _1_11_문자열_압축 {
    public static String solutionTRESH(String str) {
        StringBuilder answer = new StringBuilder();

        int p = 0;
        char prev = '\0';
        for (int i = 0; i < str.length(); i++) {
            char now = str.charAt(i);
            if (now == prev)
                p++;
            if (now != prev || i == str.length() - 1) {
                if (prev != '\0')
                    answer.append(prev);
                if (p > 1)
                    answer.append(p);
                prev = now;
                p = 1;
            }
        }

        return answer.toString();
    }

    public static String solution(String str) {
        str = str + '\0';
        StringBuilder sb = new StringBuilder();
        int cnt = 1;
        for (int i = 0; i < str.length() - 1; i++) {
            if(str.charAt(i) == str.charAt(i + 1))
                cnt++;
            else {
                sb.append(str.charAt(i));
                if(cnt > 1)
                    sb.append(cnt);
                cnt = 1;
            }
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.next();
        System.out.println(solution(input));
    }
}

File: src/main/kotlin/inflearn/kotlin/ch01/1_11 문자열 압축.kt
package inflearn.kotlin.ch01

import java.util.*

object `1_11 문자열 압축` {
    fun solution(str: String): String {
        val s = "$str "
        val sb = StringBuilder()
        var cnt = 1
        for (i in 0 until s.length - 1) {
            if (s[i] == s[i + 1]) cnt++ else {
                sb.append(s[i])
                if (cnt > 1) sb.append(cnt)
                cnt = 1
            }
        }
        return sb.toString()
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val input = sc.next()
        println(solution(input))
    }
}